moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -553,6 +554,92 @@ TEST(FK, OneRobot)
+    srdf_model.reset(new srdf::Model());
+    srdf_model->initFile(*urdf_model, res_path + "/pr2_description/srdf/robot.xml");
+    robot_model.reset(new moveit::core::RobotModel(urdf_model, srdf_model));
+  }
+
+  virtual void TearDown()
+  {
+  }
+
+protected:
+  urdf::ModelInterfaceSharedPtr urdf_model;
+  srdf::ModelSharedPtr srdf_model;
+  moveit::core::RobotModelConstPtr robot_model;
+};
+
+void generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
+                      const moveit::core::RobotModelConstPtr& robot_model,
+                      const robot_model::JointModelGroup* joint_model_group)
+{
+  traj.clear();
+
+  // 3 waypoints with default joints
+  std::shared_ptr<robot_state::RobotState> robot_state(new robot_state::RobotState(robot_model));
+  robot_state->setToDefaultValues();
+  for (std::size_t traj_ix = 0; traj_ix < 3; ++traj_ix)
+    traj.push_back(robot_state);
+
+  // 4th waypoint with a large jump of 1.01 in first joint
+  robot_state.reset(new robot_state::RobotState(*robot_state));
+  std::vector<double> joint_positions;
+  robot_state->copyJointGroupPositions(joint_model_group, joint_positions);
+  joint_positions[0] -= 1.01;
+  robot_state->setJointGroupPositions(joint_model_group, joint_positions);
+  traj.push_back(robot_state);
+}
+
+TEST_F(LoadPR2, testAbsoluteJointSpaceJump)
+{
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
+  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
+
+  // direct call of absolute version
+  generateTestTraj(traj, robot_model, joint_model_group);
+  double fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
+  EXPECT_EQ(traj.size(), 3);  // traj should be cut
+  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+
+  // indirect call
+  generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(1.0, 1.0));
+  EXPECT_EQ(traj.size(), 3);  // traj should be cut
+  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+
+  // ignore revolute joints
+  generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(0.0, 1.0));
+  EXPECT_EQ(traj.size(), 4);  // traj should not be cut
+  EXPECT_NEAR(fraction, 4. / 4., 0.01);
+}
+
+TEST_F(LoadPR2, testRelativeJointSpaceJump)
+{
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
+  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
+
+  // direct call of absolute version: factor slightly smaller than 3 (1.01 > 2.99 * 1.01 / 3)
+  generateTestTraj(traj, robot_model, joint_model_group);
+  double fraction = robot_state::RobotState::testRelativeJointSpaceJump(joint_model_group, traj, 2.99);
```

Comment:
I think this is a regression from the tests in #821. More test coverage is generally better. Why did you revert these?  
```

std::size_t generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
                             const moveit::core::RobotModelConstPtr& robot_model,
                             const robot_model::JointModelGroup* joint_model_group)
{
  traj.clear();

  std::size_t n_joints = joint_model_group->getJointModelNames().size();
  for (std::size_t traj_ix = 0; traj_ix < 3; ++traj_ix)
  {
    std::shared_ptr<robot_state::RobotState> robot_state1(new robot_state::RobotState(robot_model));
    robot_state1->setToDefaultValues();
    traj.push_back(robot_state1);
  }

  // This is a jump
  std::shared_ptr<robot_state::RobotState> robot_state2(new robot_state::RobotState(robot_model));
  robot_state2->setToDefaultValues();
  std::vector<double> joint_positions;
  robot_state2->copyJointGroupPositions(joint_model_group, joint_positions);
  joint_positions[0] -= 1.01;
  robot_state2->setJointGroupPositions(joint_model_group, joint_positions);
  traj.push_back(robot_state2);

  return traj.size();
}

TEST_F(LoadPR2, testJointSpaceJumpAbsolute)
{
  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
  std::size_t traj_len = generateTestTraj(traj, robot_model, joint_model_group);

  robot_state::JumpThreshold jt_abs(1.0, 1.0);
  double fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, jt_abs);

  EXPECT_NEAR(traj.size(), 3, 0.01);
  EXPECT_NEAR(fraction, 3. / (double)traj_len, 0.01);
}

TEST_F(LoadPR2, testJointSpaceJumpRelative)
{
  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
  std::size_t traj_len = generateTestTraj(traj, robot_model, joint_model_group);

  robot_state::JumpThreshold jt_rel(1.0);
  double fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, jt_rel);

  EXPECT_NEAR(traj.size(), 3, 0.01);
  EXPECT_NEAR(fraction, 3. / (double)traj_len, 0.01);
}
```

Reply:
As discussed [here](https://github.com/ros-planning/moveit/pull/821#discussion_r179578526), your test extensions were overly complex and didn't added any functionality. In 20dc432d63c91c2a010254284338b154f8843015 you already reverted most of those changes.
I simply cleaned them up even more to facilitate readability. 
I agree that more test coverage is better. However, I'm puzzled why you complain, as I have even more tests, namely testing for both direct (`testAbsoluteJointSpaceJump` / `testRelativeJointSpaceJump`) and indirect calls (`testJointSpaceJump`).